### PR TITLE
Slice the TSL by primitive name, provide arguments via cmake

### PR DESCRIPTION
**Pull request by alexKrauseTUD**

### CODE REVIEW

File: generator/core/tsl_generator.py [hunk 1]
```diff
@@ -111,6 +112,64 @@ def generate(self, relevant_hardware_flags: List[str] = None):
             extensions_dict = {"generated_extensions": extensions_list}
             yaml_store(config.tsl_extensions_yaml_output_path, extensions_dict)
         relevant_primitives_class_set: TSLPrimitiveClassSet = slicer.slice_primitives(self.__tsl_primitiveclass_set)
+        
+        """ Only filter if any relevant primitves are set. """
+        if config.get_config_entry("relevant_primitives"):
```
Comment (JPietrzykTUD) [inline generator/core/tsl_generator.py:117]: Please check for list containing only empty strings

File: generator/config/default_conf.yaml [hunk 2]
```diff
@@ -8,6 +8,8 @@ configuration:
    use_concepts:                  True
    emit_workaround_warnings:      True
    relevant_types:                 ["uint8_t", "int8_t", "uint16_t", "int16_t", "uint32_t", "int32_t", "uint64_t", "int64_t", "float", "double"]
+   relevant_primitives:            []
```
Comment (JPietrzykTUD) [inline generator/config/default_conf.yaml:11]: Do we need this?

File: generator/core/tsl_config.py [hunk 3]
```diff
@@ -420,6 +420,10 @@ def parse_args(**kwargs) -> dict:
     parser.add_argument('--types', default=None, nargs='*',
                         help=types_help,
                         dest='configuration:relevant_types', metavar="TYPES")
+    
+    parser.add_argument('--primitives', default=None, nargs='*',
```
Comment (JPietrzykTUD) [inline generator/core/tsl_config.py:424]: default can be [] ?

**Review by JPietrzykTUD — CHANGES_REQUESTED** [review]
Please go over the comments

File: generator/core/tsl_config.py (hunk 3 above)
Comment (alexKrauseTUD) [reply generator/core/tsl_config.py]: implemented.

File: generator/core/tsl_generator.py (hunk 1 above)
Comment (alexKrauseTUD) [reply generator/core/tsl_generator.py]: done as filtering based on string length in the config parameter.

**Review by JPietrzykTUD — APPROVED** [review]

**Pull request merged**